packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -431,28 +442,33 @@ Static group authorization should perform as expected.`
 
             // Get the directives we need to add to the GraphQL nodes
             const directives = this.getDirectivesForRules(rules, false);
+            let operationName: string = undefined;
 
             if (directives.length > 0) {
                 this.addDirectivesToField(ctx, parent.name.value, field.name.value, directives);
 
                 const addDirectivesForOperation = (operationType: ModelDirectiveOperationType) => {
                     if (modelConfiguration.shouldHave(operationType)) {
-                        const operationName = modelConfiguration.getName(operationType);
+                        operationName = modelConfiguration.getName(operationType);
                         // If the parent type has any rules for this operation AND
                         // the default provider we've to get directives including the default
                         // as well.
                         const includeDefault = this.isTypeHasRulesForOperation(parent, operationType);
                         const operationDirectives = this.getDirectivesForRules(rules, includeDefault);
 
                         this.addDirectivesToOperation(ctx, ctx.getQueryTypeName(), operationName, operationDirectives);
-                        this.addFieldToResourceReferences(ctx.getQueryTypeName(), operationName, rules);
                     }
                 }
 
                 addDirectivesForOperation('get');
                 addDirectivesForOperation('list');
             }
 
+            if (operationName) {
+                this.addFieldToResourceReferences(ctx.getQueryTypeName(), operationName, rules);
+                this.addFieldToResourceReferences(ctx.getQueryTypeName(), operationName, rules);
```

Comment:
Why do we have the same statement twice?

Reply:
Already refactored, that piece was not correct there, victim of copy and paste.